workers/src/main/scala/workers/Cache.scala

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package workers
+
+import boopickle.DefaultBasic.*
+import boopickle.Pickler
+import cats.effect.kernel.Async
+import cats.effect.kernel.Sync
+import cats.syntax.all.*
+import explore.model.boopickle.Boopickle.*
+import japgolly.scalajs.react.callback.*
+import japgolly.webapputil.indexeddb.IndexedDb
+import japgolly.webapputil.indexeddb.IndexedDb.Database
+import japgolly.webapputil.indexeddb.IndexedDbKey
+import japgolly.webapputil.indexeddb.KeyCodec
+import japgolly.webapputil.indexeddb.ObjectStoreDef
+import japgolly.webapputil.indexeddb.ValueCodec
+import org.scalajs.dom
+import org.scalajs.dom.CacheStorage
+import org.scalajs.dom.IDBFactory
+import org.scalajs.dom.{Cache => JsCache}
+
+import java.time.Instant
+import java.util.Base64
+import scala.scalajs.js.typedarray.Int8Array
+
+import scalajs.js
+import scalajs.js.JSConverters.*
+
+/**
+ * Cacheable computation.
+ */
+case class Cacheable[F[_], I, O](name: String, version: Int, invoke: I => F[O])
+
+/**
+ * Generic cache interface.
+ */
+sealed trait Cache[F[_]]:
+  def eval[I: Pickler, O: Pickler](computation: Cacheable[F, I, O]): I => F[O]
+
+/**
+ * `NoCache` is used when, for some reason, a cache could not be initialized.
+ */
+case class NoCache[F[_]]() extends Cache[F]:
```

Comment:
We are actually not checking if the API is available.

The API returns a `js.UndefOr` in both cases when trying to open a cache or a db, so this just handles that case in case opening fails.